### ISSUE TITLE
Deconstructing Single Case Discrimated Union in function arg gives erroneous warning

### ISSUE THREAD

**Issue by cljponies** [fsprojects/FSharpLint#97]
Say I have:

``` F#
type SingleCaseDU = SingleCaseDU of int
let extractInt (SingleCaseDU myInt) = 
  myInt

let singleCaseDU = SingleCaseDU 5

let result = extractInt singleCaseDU
```

FsharpLint v0.2.1 erroneously gives the following warning:
`Expected camel case identifier but was SingleCaseDU`
with the squiggle in the following line:
`let extractInt (SingleCaseDU myInt) =`

**Comment by duckmatt** [comment]
Thanks for the report, I'll look into getting this fixed asap and releasing a new nuget package with the fix

**Comment by duckmatt** [comment]
This is fixed in [0.2.2](https://www.nuget.org/packages/FSharpLint/)

**Comment by kerams** [comment]
Hi, I'm still seeing this in the linter bundled with the latest version of VFPT.

```fsharp
module String10 =
    type T = private | String10 of string

    let create str = String10 str

    let value (String10 str) = str
```
Removing the `private` part does not help.

**Comment by duckmatt** [comment]
Hi @kerams, thanks for reporting this I'll try repeating with the latest version of the linter

**Comment by duckmatt** [comment]
Closing as unable to repeat with regression test
